modules/services/attic.nix

```diff
@@ -0,0 +1,95 @@
+{ pkgs, lib, nglib, config, ... }:
+let
+  cfg = config.services.attic;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.attic = {
+    enable = lib.mkEnableOption "attic";
+    package = lib.mkPackageOption pkgs "attic-server" { };
+
+    user = lib.mkOption {
+      description = "attic user";
+      type = lib.types.str;
+      default = "atticd";
+    };
+
+    group = lib.mkOption {
+      description = "attic group";
+      type = lib.types.str;
+      default = "atticd";
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          storage = {
+            type = lib.mkOption {
+              type = lib.types.enum [ "local" "s3" ];
+            };
+
+            path = lib.mkOption {
+              type = lib.types.str;
```

Comment:
This could be `types.path`

Reply:
Like I said above, the type can be `s3` as well so we cannot restict the path to be a file system path only.

Reply:
Oh sorry that I missed.